src/jvm/backtype/storm/serialization/SerializationFactory.java

```
@@ -32,7 +33,25 @@ public static Kryo getKryo(Map conf) {
         IKryoFactory kryoFactory = (IKryoFactory) Utils.newInstance((String) conf.get(Config.TOPOLOGY_KRYO_FACTORY));
         Kryo k = kryoFactory.getKryo(conf);        
         k.register(byte[].class);
-        k.register(ListDelegate.class);
+
+        /* tuple payload serializer could be specified via configuration */
+        String payloadSerializerName = (String)conf.get(Config.TOPOLOGY_TUPLE_SERIALIZER);
+        if (payloadSerializerName==null) 
+            k.register(ListDelegate.class, new ListDelegateSerializer()); //use default payload serializer
+        else {
+            try {
+                Class serializerClass  = Class.forName(payloadSerializerName);
```

If the class isn't found there should be a runtime error. The user has done something wrong in this case and should be notified – silently using the default serializer is confusing and could be dangerous (e.g., the user was expecting encryption).